cmd/configure/attributes/pod/pod.go

```diff
@@ -63,53 +74,32 @@ func ParseAttributes(rawAttributes []string) (Attributes, error) {
 		return Attributes{}, err
 	}
 
-	err = filterOutUserDefined(rawMap, result)
-	if err != nil {
-		return Attributes{}, err
-	}
-
 	result.UserDefined = rawMap
 
 	return result, nil
 }
 
-func filterOutUserDefined(rawInput map[string]string, parsedInput Attributes) error {
-	parsedMap, err := parsedInput.ToMap()
-	if err != nil {
-		return err
-	}
-
-	for key := range parsedMap {
-		delete(rawInput, key)
-	}
-
-	return nil
-}
-
-// ToArgs is a helper func to convert an pod.Attributes to a list of args that can be put into a Pod Template
+// ToArgs is a helper func to convert a pod.Attributes to a list of args that can be put into a Pod Template.
 func ToArgs(attributes Attributes) ([]string, error) {
-	var args []string
+	var args = make([]string, 0, len(attributes.UserDefined))
 
-	attrMap, err := attributes.ToMap()
-	if err != nil {
-		return nil, err
-	}
-
-	for key, value := range attrMap {
-		if value == "" {
-			continue
-		}
-
-		args = append(args, fmt.Sprintf("--%s=%s=%s", Flag, key, value))
-	}
-
-	for key, value := range attributes.UserDefined {
-		if value == "" {
-			continue
-		}
-
-		args = append(args, fmt.Sprintf("--%s=%s=%s", Flag, key, value))
+	for k, v := range attributes.UserDefined {
+		args = append(args, fmt.Sprintf("--%s=%s", k, v))
 	}
 
 	return args, nil
 }
+
+// filterOutUserDefined removes known structured fields from the raw map, leaving only user-defined fields.
+func filterOutUserDefined(raw map[string]string, _ Attributes) {
+	// Remove known structured fields from raw map
+	delete(raw, "k8s.pod.name")
+	delete(raw, "k8s.pod.uid")
+	delete(raw, "k8s.node.name")
+	delete(raw, "k8s.namespace.name")
+	delete(raw, "k8s.workload.kind")
+	delete(raw, "k8s.workload.name")
+	delete(raw, "k8s.cluster.uid")
+	delete(raw, "k8s.cluster.name")
+	delete(raw, "dt.entity.kubernetes_cluster")
+}
```

Comment:
how is this better than `filterOutUserDefined` ?

you even pass `_ Attributes` for what reason